neasqc_wp61/models/quantum/circuit.py

```diff
@@ -1,101 +1,110 @@
 """
 Circuit
 =======
-Module containing the base class for the variational circuits of Alpha3 model
+Module containing the base class for the variational circuits of Alpha3 model.
 
 """
 from abc import ABC, abstractmethod
-from typing import Callable
+from typing import Callable, Dict
 
 import pennylane as qml
-from pennylane.measurements.expval import ExpectationMP
 import torch
 
+
 class Circuit(ABC):
     """
-    Base class for circuits
+    Base class for circuits.
     """
     def __init__(
         self, n_qubits : int,  n_layers : int,
-        axis_embedding : str, observables : list[qml.operation.Operator],
-        device_name : str = "default.qubit", data_rescaling : Callable = None,
-        **kwargs
+        axis_embedding : str, observables : Dict[int, qml.operation.Operator],
+        output_probabilities : bool, device_name : str = "default.qubit",
```

Comment:
I would give a default value to this output_prob boolean. Either tru or false, whatever we use most often.